### PR TITLE
Add Nutanix-specific customizations to manifests

### PR DESCRIPTION
**Pull request by thunderboltsid**
- Update templates in data/data/manifests
- Add nutanix platform customizations in pkg/asset/manifests

### CODE REVIEW

File: data/data/manifests/openshift/cloud-creds-secret.yaml.template [hunk 1]
```diff
@@ -45,4 +47,9 @@ data:
   ovirt_password: {{.CloudCreds.Ovirt.Base64encodePassword}}
   ovirt_insecure: {{.CloudCreds.Ovirt.Base64encodeInsecure}}
   ovirt_ca_bundle: {{.CloudCreds.Ovirt.Base64encodeCABundle}}
+{{- else if .CloudCreds.Nutanix}}
```
Comment (rvanderp3) [inline data/data/manifests/openshift/cloud-creds-secret.yaml.template:50]: What is, or is there, a relationship between this secret and the CSI driver operator secret `ntnx-secret`?  The [enhancement](https://github.com/openshift/enhancements/blob/8e71d3860e433fa5d118f31c69825df5913da7c1/enhancements/installer/nutanix-ahv-ipi.md?plain=1#L124) references the Nutanix CSI driver operator which depends on `ntnx-secret` in the `ntnx-system` namespace.

Comment (thunderboltsid) [reply data/data/manifests/openshift/cloud-creds-secret.yaml.template:50]: From my understanding, these secrets have the same contents albeit in the case of CSI driver they tend to follow a string based encoding of the values (prism-element-ip:prism-port:username:password).

Comment (rvanderp3) [reply data/data/manifests/openshift/cloud-creds-secret.yaml.template:50]: @thunderboltsid I think this the [discussion](https://github.com/openshift/cloud-credential-operator/pull/450#discussion_r800788502) you are referring to.  Do you happen to have a timeline on when the transition to moving the configuration of the ip/port to the infrastructure CR might take place and would moving the configuration to the infrastructure CR at a later time cause issues?

Comment (thunderboltsid) [reply data/data/manifests/openshift/cloud-creds-secret.yaml.template:50]: We can move the IP/Port over to infra CR at a later date without any issues.

File: pkg/asset/manifests/cloudproviderconfig.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 	}
 
 	switch installConfig.Config.Platform.Name() {
-	case libvirttypes.Name, nonetypes.Name, baremetaltypes.Name, ovirttypes.Name:
+	case libvirttypes.Name, nonetypes.Name, baremetaltypes.Name, ovirttypes.Name, nutanixtypes.Name:
```
Comment (staebler) [inline pkg/asset/manifests/cloudproviderconfig.go:96]: I am a bit surprised that there is not a cloud provider config. Is the name of the secret with the creds hard-coded somewhere?

File: pkg/asset/manifests/template.go [hunk 3]
```diff
@@ -54,6 +54,14 @@ type OvirtCredsSecretData struct {
 	Base64encodeCABundle string
 }
 
+// NutanixCredsSecretData holds encoded credentials and is used to generate cloud-creds secret
+type NutanixCredsSecretData struct {
+	Base64encodeNutanixEndpoint string
+	Base64encodeNutanixPort     string
+	Base64encodeNutanixUsername string
+	Base64encodeNutanixPassword string
```
Comment (staebler) [inline pkg/asset/manifests/template.go:62]: "Nutanix" in these names is redundant.
```suggestion
	Base64encodeEndpoint string
	Base64encodePort     string
	Base64encodeUsername string
	Base64encodePassword string
```

File: pkg/asset/manifests/cloudproviderconfig.go (hunk 2 above)
Comment (thunderboltsid) [reply pkg/asset/manifests/cloudproviderconfig.go:96]: Currently, yes. `nutanix-credentials` is the credential name.

File: data/data/manifests/openshift/cloud-creds-secret.yaml.template [hunk 4]
```diff
@@ -45,4 +47,9 @@ data:
...
+  NUTANIX_ENDPOINT: {{.CloudCreds.Nutanix.Base64encodeNutanixEndpoint}}
+  NUTANIX_PORT: {{.CloudCreds.Nutanix.Base64encodeNutanixPort}}
+  NUTANIX_USER: {{.CloudCreds.Nutanix.Base64encodeNutanixUsername}}
+  NUTANIX_PASSWORD: {{.CloudCreds.Nutanix.Base64encodeNutanixPassword}}
```
Comment (rvanderp3) [inline data/data/manifests/openshift/cloud-creds-secret.yaml.template:54]: @thunderboltsid does this need to refactored in light of https://github.com/openshift/installer/pull/5631/files#diff-89d415ff2b74dae9eb1035ae2b655b8f099235198e424c4108e000601cc1dec6R218 ?

Comment (thunderboltsid) [reply data/data/manifests/openshift/cloud-creds-secret.yaml.template:54]: Indeed

**Pull request merged**